src/ICSharpCode.SharpZipLib/Zip/ZipOutputStream.cs

```diff
@@ -372,7 +379,15 @@ public void PutNextEntry(ZipEntry entry)
 				if (headerInfoAvailable)
 				{
 					ed.AddLeLong(entry.Size);
-					ed.AddLeLong(entry.CompressedSize);
+
+					long entryCompressedSize = entry.CompressedSize;
+
+					if (entry.IsCrypted)
+					{
+						entryCompressedSize += entry.EncryptionOverheadSize;
+					}
+
+					ed.AddLeLong(entryCompressedSize);
```

Comment:
Could be simplified (since `IsCrypted` is already checked in `EncryptionOverheadSize`)
```suggestion
					ed.AddLeLong(entry.CompressedSize + entry.EncryptionOverheadSize);
```